src/Raven.Server/Documents/Indexes/Persistence/IndexOperationBase.cs

```diff
@@ -113,16 +113,26 @@ protected static int LuceneGetPageSize(IndexSearcher searcher, long pageSize)
         return (int)pageSize;
     }
 
-    protected static int CoraxGetPageSize(global::Corax.IndexSearcher searcher, int bufferSize, IndexQueryServerSide query)
+    protected static int CoraxGetPageSize(global::Corax.IndexSearcher searcher, int pageSize, IndexQueryServerSide query, bool isBoolean = false)
     {
-        var size = searcher.NumberOfEntries;
-        if (size > int.MaxValue)
+        var numberOfEntries = searcher.NumberOfEntries;
+        
+        if (numberOfEntries == 0)
+            return 16;
+
+        if (pageSize <= 0)
+            return (int)numberOfEntries;
```

Comment:
What happens if we have > 2B entries, we'll get negative value here.

Reply:
Fixed.